tools/lint/lib/checks/license.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from ..check import Check
+
+_MIN_YEAR = 2009
+_MAX_YEAR = 2030
+
+_LICENSE_PATTERN = re.compile(
+    r'\/\/ Copyright( \([cC]\))? (\w+) .+\. {1,2}All rights reserved\.[\r\n]{1,2}' +
+    r'(' +
+        r'\/\/ (' +
+            r'This code is governed by the( BSD)? license found in the LICENSE file\.' +
+            r'|' +
+            r'See LICENSE for details' +
+        r')' +
+        r'|' +
+        r'\/\/ Use of this source code is governed by a BSD-style license that can be[\r\n]{1,2}' +
+        r'\/\/ found in the LICENSE file\.' +
+        r'|' +
+        r'\/\/ See LICENSE or https://github\.com/tc39/test262/blob/master/LICENSE' +
+    r')', re.IGNORECASE)
+
+class CheckLicense(Check):
+    '''Ensure tests declare valid license information.'''
+    ID = 'LICENSE'
+
+    def run(self, name, meta, source):
+        if meta and 'flags' in meta and 'generated' in meta['flags']:
+            return
+
+        match = _LICENSE_PATTERN.search(source)
+
+        if not match:
+            return 'No license information found.'
```

Comment:
I'm positive #851 is on the way to reach final consensus soon, potentially in the next meeting. So I would avoid having this check. It's good to fix the current tests, but hopefully an unnecessary check for future test files.